web/components/admin/TextField.tsx

```diff
@@ -53,14 +54,29 @@ export const TextField: FC<TextFieldProps> = ({
   useTrim,
   value,
 }) => {
+  const [hasPwdChanged, setHasPwdChanged] = useState(false);
+  const [showPwdButton, setShowPwdButton] = useState(false);
+  const regex = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#$%^&*]).{8,192}$/;
+  const [form] = Form.useForm();
   const handleChange = (e: any) => {
     // if an extra onChange handler was sent in as a prop, let's run that too.
     if (onChange) {
       const val = type === TEXTFIELD_TYPE_NUMBER ? e : e.target.value;
+      setShowPwdButton(true);
+      if (fieldName === 'adminPassword' && regex.test(val)) {
```

Comment:
Looks like there's still hardcoded references to an implementation of the text fields in the component itself with `adminPassword`. The generic components shouldn't have to know about specific instances of them.